Multiphase sulfate from Travis et al., 2025

ktravis213 opened the pull request:
### Name and Institution (Required)

Name: Katherine Travis
Institution: NASA LaRC

### Describe the update

Please provide a clear and concise overview of the update.

This pull request is associated with the reference below. It implements multiphase (heterogeneous) sulfate chemistry in KPP. This includes SO2 oxidation by NO2, O3, H2O2, TMI-catalyzed O2, and HCHO.  All oxidants but NO2 have a dependence on ionic strength calculated in HETP. Diagnostics are included for the uptake coefficient gamma in StateChm and production of sulfate from each new pathway.  There is a separate pull request for the addition of the ionic strength diagnostic in HETP.

### Expected changes

Please provide details on how this update will impact model output and include plots or tables as needed.
Expected changes are increased sulfate and HMS and reduced SO2. The plot below shows the global increase in sulfate from a 2x25 run for January 2016. 
<img width="1267" height="308" alt="JanuaryPSO4" src="https://github.com/user-attachments/assets/4059fcb6-9ff5-4f81-8877-af7d8a78cab7" />

### Reference(s)

- *Year-Round Analysis of Multiphase Sulfate Production in Aerosol Particles in East Asia*, Katherine R. Travis. Benjamin A. Nault, James H. Crawford, Hwajin Kim, Qi Chen, Yan Zheng, Tengyu Liu, Jose L. Jimenez, Pedro Campuzano-Jost, Paul O. Wennberg, John D. Crounse, L. Gregory Huey, <u>ACS ES&T Air</u>, submitted, 2025, DOI: [10.1021/acsestair.5c00136](https://doi.org/10.1021/acsestair.5c00136)

### Related Github Issue

- Merge concurrently with https://github.com/geoschem/HETerogeneous-vectorized-or-Parallel/pull/4

Commit:
Addition of multiphase sulfate chemistry according to https://doi.org/10.1021/acsetair.5c00136. Includes SO2 oxidation in aerosol water by H2O2, O3, NO2, HCHO, and TMI-catalzyed O2. Addition of cloud HPLUS diagnostic since cloud pH diagnostic appears broken.

Signed-off-by: Katherine R. Travis <ktravis1@discover35.prv.cube>

Commit:
Addition of multiphase sulfate chemistry according to https://doi.org/10.1021/acsetair.5c00136. Includes SO2 oxidation in aerosol water by H2O2, O3, NO2, HCHO, and TMI-catalzyed O2. Addition of cloud HPLUS diagnostic since cloud pH diagnostic appears broken.

Signed-off-by: Katherine R. Travis <ktravis1@discover35.prv.cube>

Commit:
Addition of multiphase sulfate chemistry according to https://doi.org/10.1021/acsetair.5c00136. Includes SO2 oxidation in aerosol water by H2O2, O3, NO2, HCHO, and TMI-catalzyed O2. Addition of cloud HPLUS diagnostic since cloud pH diagnostic appears broken.

Signed-off-by: Katherine R. Travis <ktravis1@discover35.prv.cube>

yantosca commented:
Thanks @ktravis213, we'll look at this shortly.  It should be able to go into 14.7.0.

yantosca reviewed (changes requested):
Hi @ktravis213, thanks for this PR.  I've noted some items to be addressed (and a couple of questions).  

Also, I see that there are several new family species added.  Would you know if this increases the run time?

Commit:
Edits to original pull request.
--> Change 'uptake probability' to 'sticking coefficient' for consistency
--> Consolidate production trackers from 13 to 3 (PSO4MP for multiphase sulfate, and PHMSAQ and PHMSMP for HMS aqueous and multiphase production)
--> Improved computational efficiency by removing ELSE blocks.
--> Added _dp to constants
--> Add Ionic strength diagnostic to GCHP HISTORY template

Signed-off-by: Katherine R. Travis <ktravis1@discover36.prv.cube>

yantosca commented:
Hi @ktravis213, I have rebased your code atop 14.7.0.  I'm opening a new PR #3202 and will close this one.  We will get this update into 14.8.0.
